modules/unpickler/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/Scala3Unpickler.scala

```diff
@@ -316,15 +320,37 @@ class Scala3Unpickler(
     else encodedScalaName == expectedName
 
   private def matchSignature(method: binary.Method, symbol: TermSymbol): Boolean =
-    symbol.signedName match
-      case SignedName(_, sig, _) =>
-        matchArguments(sig.paramsSig, method.declaredParams)
-        && method.declaredReturnType.forall(matchType(sig.resSig, _))
-      case _ =>
-        // TODO compare symbol.declaredType
-        method.declaredParams.isEmpty
+    def parametersName(tpe: TypeOrMethodic): List[String] =
+      tpe match
+        case t: MethodType =>
+          t.paramNames.map(_.toString()) ++ parametersName(t.resultType)
+        case t: PolyType =>
+          parametersName(t.resultType)
+        case _ => List()
+
+    def matchesCapture(paramName: String) =
+      val pattern = ".+\\$\\d+".r
+      pattern.matches(
+        paramName
+      ) || (method.isExtensionMethod && paramName == "$this") || (method.isClassInitializer && paramName == "$outer")
+
+    val paramNames: List[String] = parametersName(symbol.declaredType)
+    val capturedParams = method.allParameters.size - paramNames.size
+    if !method.allParameters.take(capturedParams).map(_.name).forall(matchesCapture) then false
+    else
+      symbol.signedName match
+        case SignedName(_, sig, _) =>
+          val declaredParameters = method.allParameters.takeRight(method.allParameters.size - capturedParams)
+          matchArgumentsTypes(sig.paramsSig, declaredParameters)
+          && declaredParameters.map(_.name).zip(paramNames).forall((s1, s2) => s1 == s2)
```

Comment:
`.corresponds` also check the size

```suggestion
          && declaredParameters.map(_.name).corresponds(paramNames)((n1, n2) => n1 == n2)
```